airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/operation/SnowflakeStagingClient.kt

```diff
@@ -63,8 +79,21 @@ class SnowflakeStagingClient(private val database: JdbcDatabase) {
         recordsData: SerializableBuffer
     ) {
         val query = getPutQuery(stageName, stagingPath, recordsData.file!!.absolutePath)
-        log.info { "Executing query: $query" }
-        database.execute(query)
+        val queryId = UUID.randomUUID()
+        log.info { "query $queryId, executing $query" }
+        val results = database.queryJsons(query)
+        if (
+            results.isNotEmpty() &&
+                (results.first().has("source_size") && results.first().has("target_size"))
+        ) {
+            if (results.first().get("source_size").asLong() == 0L) {
+                // TODO: Should we break the Sync rather than proceeding with empty file for COPY ?
```

Comment:
imo eventually yes. would be good to release with just the log for now, and if https://github.com/airbytehq/airbyte/pull/39106 stops these log messages, then we should switch to throwing an error.